packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardFilterBuilder.tsx

```diff
@@ -230,8 +230,10 @@ export const CloudPulseDashboardFilterBuilder = React.memo(
     );
 
     const toggleShowFilter = () => {
-      setShowFilter((showFilterPrev) => !showFilterPrev);
-      handleToggleAppliedFilter(showFilter);
+      setShowFilter((showFilterPrev) => {
+        handleToggleAppliedFilter(showFilterPrev);
```

Comment:
this is not correct right

        handleToggleAppliedFilter(!showFilterPrev);

Reply:
no. if showFilterPrev is true that means previously filter was visible & now it is going to be false that means applied filter should be visible